core/src/main/scala/mill/main/ParseArgs.scala

```diff
@@ -0,0 +1,104 @@
+package mill.main
+
+import mill.discover.Mirror
+import mill.util.EitherOps
+import fastparse.all._
+
+object ParseArgs {
+
+  def apply(scriptArgs: Seq[String])
+    : Either[String, (List[List[Mirror.Segment]], Seq[String])] = {
+    val (selectorStrings, args) = extractSelsAndArgs(scriptArgs)
+    for {
+      _ <- validateSelectors(selectorStrings)
+      expandedSelectors <- EitherOps
+        .sequence(selectorStrings.map(expandBraces))
+        .map(_.flatten)
+      selectors <- EitherOps.sequence(expandedSelectors.map(extractSegments))
+    } yield (selectors.toList, args)
+  }
+
+  def extractSelsAndArgs(
+      scriptArgs: Seq[String]): (Seq[String], Seq[String]) = {
+    val multiFlags = Seq("--all", "--seq")
+    val isMultiSelectors = scriptArgs.exists(multiFlags.contains)
```

Comment:
This doesn't seem to handle the case where `--all` is an argument that's meant to be passed to the command being run:

```diff
lihaoyi mill$ git diff
diff --git a/build.sc b/build.sc
index 16642f6..e9e7c90 100755
--- a/build.sc
+++ b/build.sc
@@ -7,6 +7,9 @@ import mill._
 import mill.scalalib._, publish._
 import mill.modules.Jvm.createAssembly

+def foo(x: String*) = T.command{
+  println(x)
+}
```
```
lihaoyi mill$ bin/target/mill foo --bar
Compiling /Users/lihaoyi/Dropbox/Workspace/mill/build.sc
[1/1] foo
ArrayBuffer(--bar)
lihaoyi mill$ bin/target/mill foo --all
[1/1] foo
ArrayBuffer()
```

Would checking for the first element in `scriptArgs` be enough to solve that? e.g. for cases like `mill core.test --all scalalib.test`, we'd expect `--all scalalib.test` to be passed as arguments to `core.test`.

(We should also have a test for this)

Reply:
Sure, will cover this case